docs: document expectations around AI use

kirkwaiblinger opened the pull request:
## PR Checklist

- [x] Addresses an existing open issue: fixes #11416 
- [x] That issue was marked as [accepting prs](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aopen+is%3Aissue+label%3A%22accepting+prs%22)
- [x] Steps in [Contributing](https://typescript-eslint.io/contributing) were taken

## Overview

I've just put up an initial message; iteration and criticism is very much invited! 

@typescript-eslint/triage-team

bradzacher commented on docs/contributing/Pull_Requests.mdx:
```diff
@@ -25,6 +25,23 @@ Please don't:
+
+While we cannot and will not attempt to ban contributions which make use of AI, we ask that you use AI responsibly:
+
+- Always review AI-generated content closely
```
I'd also suggest adding this:

```suggestion
- Always review AI-generated content closely **_before submitting a PR._**
```

To emphasise that we don't want people to vibe a change, ping us with a PR creation and then review it later.

kirkwaiblinger replied:
Added "before submitting a PR". 

Re some of the language like "to champion".... I'm at a loss for a good idea to replace that. I want to convey that you should not only be able to do the feature, but also defend it, incorporate fixes/changes, and foresee potential impacts such as breaking changes or important interactions with other features. I lifted the "champion" wording from some of our canned responses on issues, see 

https://github.com/typescript-eslint/typescript-eslint/blob/635133a293cbea63447f7b579e8c51306e01cd7f/.github/replies.yml#L66-L71

Open to suggestions

bradzacher commented on docs/contributing/Pull_Requests.mdx:
```diff
@@ -25,6 +25,23 @@ Please don't:
+- Only use AI for contributions that you would understand well enough to champion and respond to feedback on without making use of AI
+- Do not ignore our issue and PR templates
+
+Don't let this dissuade you from contributing to typescript-eslint! We are generally more than happy to assist new contributors and help them improve at our repo. We just are not interested in babysitting anyone's LLM instances.
```
Good call out yeah.

If the persons entire role in the process is agentic middleman then we ask that they remove themselves from the process entirely and just not submit the PR.

We can talk to the agent without them as a go-between.

JoshuaKGoldberg commented on docs/contributing/Pull_Requests.mdx:
```diff
@@ -25,6 +25,28 @@ Please don't:
 - Comment on a closed PR
   - Reasoning: It is much easier for maintainers to not lose track of things if they are posted as issues. If you think there's a bug in typescript-eslint, the right way to ask is to [file a new issue](https://github.com/typescript-eslint/typescript-eslint/issues/new/choose). The issue templates include helpful & necessary practices such as making sure you're on the latest version of all our pack ...
 
+:::warning Contribution standards in the era of AI
```
Ehh, I'm going back and forth on this. I think we probably don't need to say too much here. And we already have a lot of docs pages... @kirkwaiblinger I don't want to block on this at all - wdyt?

kirkwaiblinger replied:
I agree that the `:::warning` block is a bit awkward. And a more permanent linking target might be preferable. Lemme poke around with this

kirkwaiblinger commented:
I've moved the AI Policy to its own page. As part of that I've changed some of the wording again.... Since it's no longer just a section on the Pull Requests page, I thought it would make sense to generalize to include "don't send us crappy AI-generated gh issues" too. 

So, that's to say, definitely inviting another round of feedback on the details and wording.

bradzacher commented on docs/contributing/AI_Contribution_Policy.mdx:
```diff
@@ -0,0 +1,25 @@
+
+While we cannot and will not attempt to ban contributions which make use of AI, we ask that you use AI responsibly:
+
+1. Always review AI-generated code closely before submitting a PR.
```
I definitely think so - OXC has added a similar clause. A Disclosure requirement is good so we can understand how much scrutiny to place over code. We can probably start to understand the mistakes that different models commonly make too.

bradzacher replied:
It's nice because it allows you to  better frame feedback.

Like if someone says "I primarily used Claude to create this PR" then my thinking would be "it's less likely they have a good understanding of this code - so I should be very specific with my feedback so they can go and make the right change". But if there is no disclosure you might assume the opposite "they have a good understanding of the code so a quick, vague comment is enough of a prompt for them to go and fix this".

It lets you set the tone and such.

Same as if you're reviewing a PR from a first time contributor versus one from another maintainer. The former you would tailor your review to be forgiving and friendly. The latter you would be a lot more "short-hand".

kirkwaiblinger replied:
Let's punt this to a followup so that we can get this out sooner and afterwards bikeshed whether such a disclosure (if we have consensus to require one) should be in the issue/pr templates or on the website only.
